compiler/src/dmd/initsem.d

```diff
@@ -1514,3 +1422,129 @@ private bool hasNonConstPointers(Expression e)
     }
     return false;
 }
+
+/**
+Given the names and values of a `StructInitializer` or `CallExp`,
+resolve it to a list of expressions to construct a `StructLiteralExp`.
+
+Params:
+    sd = struct
+    t = type of struct (potentially including qualifiers such as `const` `immutable`)
+    sc = scope of the expression initializing the struct
+    iloc = location of expression initializing the struct
+    names = identifiers passed in argument list, `null` entries for positional arguments
+    getExp = function that, given an index into `names` and destination type, returns the initializing expression
+    getLoc = function that, given an index into `names`, returns a location for error messages
+Returns: list of expressions ordered to the struct's fields, or `null` on error
+*/
+Expressions* resolveStructLiteralNamedArgs(
```

Comment:
I find this style uncommon for the dmd code base. Typically, the first parameter is on the same line with the function name, however, I don't have any strong feelings towards this.

Reply:
I changed it